uaclient/actions.py

```diff
@@ -218,3 +219,49 @@ def collect_logs(cfg: config.UAConfig, output_dir: str):
             system.write_file(
                 os.path.join(output_dir, os.path.basename(f)), content
             )
+
+
+def get_cloud_instance(
```

Comment:
This makes all sense for me, but then, to avoid duplication, let's make the `action_auto_attach` function to call this, instead of having the same logic there.